src/vs/workbench/contrib/chat/electron-sandbox/actions/voiceChatActions.ts

```diff
@@ -720,6 +721,8 @@ class BaseStopListeningAction extends Action2 {
 	}
 
 	async run(accessor: ServicesAccessor, context?: IChatExecuteActionContext): Promise<void> {
+		const accessibilitySignalService = accessor.get(IAccessibilitySignalService);
+		accessibilitySignalService.playSignal(AccessibilitySignal.voiceRecordingStopped, { allowManyInParallel: true });
```

Comment:
How come this code is not triggered and doing the right thing:

https://github.com/microsoft/vscode/blob/a307d64354046f1dab8153a2f95628da4f037d75/src/vs/workbench/contrib/speech/browser/speechService.ts#L157

Reply:
Because that is triggered but not in response to a user gesture
<img width="1591" alt="Screenshot 2024-03-26 at 7 28 49 AM" src="https://github.com/microsoft/vscode/assets/29464607/8cb88063-7f6e-4a60-a299-17edc4702185">

Reply:
call stack

<img width="653" alt="Screenshot 2024-03-26 at 7 29 33 AM" src="https://github.com/microsoft/vscode/assets/29464607/926f3514-6770-41d8-bc30-b54966afd8e3">

Reply:
But how is the audio cue on start working? Its also from a delayed event:

https://github.com/microsoft/vscode/blob/9d9f86616344823af8c9151f515e739ed9d45be0/src/vs/workbench/contrib/speech/browser/speechService.ts#L200